### DISCUSSION TITLE
Support adding product to multiple categories

### DISCUSSION THREAD

**Discussion by khalibloo** [saleor/saleor#10414] [Ideas]
### What I'm trying to achieve
We sometimes run into situations where a single product can fall into multiple categories. Depending on how the category tree has been modeled, the categories may not even be closely related. For example, a laptop bag may be rightly classified under computer/electronics as well as under school bags. Other common examples are fashion products that belong in both "men" and "women" categories.
Related: https://github.com/mirumee/saleor/issues/278

### Describe a proposed solution
Allow a product to belong to multiple categories.

### Other solutions I've tried and won't work
Collections do not quite fit this situation. When a user browses through a store, categories have the advantage of being organized in a tree structure, and therefore easier to understand.

**Comment by karolkielecki** [comment]
Hi there! Thank you for your suggestion. You can assign only one category to one product for now. We will discuss your feedback. Thanks!

**Comment by maarcingebala** [comment]
It's not on our nearest roadmap but I think we'll need it anyway at some point.

**Comment by amesas** [comment]
Is this "some point" in a near horizon? :P

**Comment by khalibloo** [comment]
For my newer saleor projects, I'm ditching categories for the storefronts. I use collections and navigation menus to achieve the same hierarchical arrangement as categories, but with the flexibility of being able to assign products to multiple collections.

It is a chore to have to manually make large navigation menus, though. There's a bug on the dashboard where the menu items "fight back" whenever you try to sort them. So I often just do this from the django shell.

**Comment by amesas** [comment]
We use custom storefront, and our categories tree is "complex".

Right now trying to figure how to import and sync more than 150k products.

**Comment by Izayda** [comment]
> We use custom storefront, and our categories tree is "complex".
> 
> 
> 
> Right now trying to figure how to import and sync more than 150k products.

How did you realize that?

**Comment by amesas** [comment]
@Izayda still not solved, when i refer to our categories tree, i'm talking about our real catalogue, not Saleor's one.

We need to find a way to mimic the same hierarchical structure. Maybe using collections metadata and setting a "parent collection". But performance is a concern here.

**Comment by gorjan-mishevski** [comment]
Its a shame this is not present given there are discussions from 2015. Most of all e commerce sites have this. Such a basic and crucial feature. Maybe its quite difficult to add it on the existing saleor core.

**Comment by narayanpromax** [comment]
This is a basic feature for all e-Commerce. one to many mapping need to be done from 1 product to multiple categories in DB

**Comment by TurkaljGoran** [comment]
Hello, 

is there any update on this, possibly on the horizon?

**Comment by tiendq** [comment]
> For example, a laptop bag may be rightly classified under computer/electronics as well as under school bags.

I think products belong to one category is fine, if _sometimes_ you need more then use collections. It's unnecessary complicated to support a product belongs to both category and collection.

**Reply by khalibloo** [reply]
The requirement comes from a business/SEO perspective. If a user/crawler goes to computer>accessories, you want to show a certain product. And you may want the same product to appear under fashion>bags. You don't know which path a user might take to arrive at your product, so you can't choose 1 category over the other.

Yes, it is technically a big challenge, but I think it's a valid requirement. I don't think just using collections solves the problem because users will navigate your site via category trees, not collections (they don't have navigation trees after all).

**Reply by tiendq** [reply]
I also think it's a valid requirement and I didn't mean collections solve it straightly. Since it's _sometimes_ you might need to customize existing features to fit your needs.

**Comment by timuric** [comment]
We have collections that are `one to many` and categories that are `one to one` which can be a useful property in some cases. If you wish to organize collections in hierarchy you can use setup a Navigation using collections in configuration.

**Comment by darrylmorley** [comment]
Has there been any movement on this, if not is there any planned movement on this?
